Lib/test/test_imghdr.py

```diff
@@ -79,7 +80,7 @@ def test_bad_args(self):
             imghdr.what()
         with self.assertRaises(AttributeError):
             imghdr.what(None)
-        with self.assertRaises(TypeError):
+        with self.assertRaises(Exception):#TypeError
```

Comment:
I don't understand this. Why changing the `TypeError` to a general `Exception`?

Reply:
As now `h.startswith(b'\xff\xd8')` raises `AttributeError` instead, you need to assert on that:

```python
with self.assertRaises(AttributeError):
```

If you do this change, all tests in `test_imghdr` should pass.